### PR TITLE
Changed the maximum factor default value to 12 in One Variable Summary dialog

### LINKED ISSUE #7825: Maximum factor shown option in one variable summarise dialog
The default value for Maximum factor shown is 7. 
I suggest this to be changed to 12. Reason being, in many of our climate workshops we often want to also have a summary of the 12 months.

1 variable

### PR DESCRIPTION
**Pull request by Vitalis95**
Fixes #7825 
@rdstern @N-thony @lloyddewit , I changed default value to 12 in Maximum Factor control

### CODE REVIEW

File: instat/dlgOneVariableSummarise.vb [hunk 1]
```diff
@@ -60,7 +60,7 @@ Public Class dlgOneVariableSummarise
         ucrReceiverOneVarSummarise.SetMeAsReceiver()
 
         ucrNudMaxSum.SetParameter(New RParameter("maxsum", 2))
-        ucrNudMaxSum.SetRDefault("7")
+        ucrNudMaxSum.SetMinMax(12, Integer.MaxValue)
```
Comment (lloyddewit) [inline instat/dlgOneVariableSummarise.vb:63]: We're hardcoding '12' several times in this dialog. If we want to change this number later, then we'll need to change it in multiple places.
Would it be better to declare a constant (e.g. `Private Const iMaxSum = 12` and use this instead?
```suggestion
        ucrNudMaxSum.SetMinMax(iMaxSum, Integer.MaxValue)
```

File: instat/dlgOneVariableSummarise.vb [hunk 2]
```diff
@@ -69,7 +69,8 @@ Public Class dlgOneVariableSummarise
         ucrPnlSummaries.AddParameterValuesCondition(rdoCustomised, "checked_radio", "customised")
         ucrPnlSummaries.AddParameterValuesCondition(rdoDefault, "checked_radio", "defaults")
         ucrPnlSummaries.AddParameterValuesCondition(rdoSkim, "checked_radio", "skim")
-        ucrPnlSummaries.AddToLinkedControls(ucrNudMaxSum, {rdoDefault}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True)
+        ucrPnlSummaries.AddToLinkedControls(ucrNudMaxSum, {rdoDefault}, bNewLinkedAddRemoveParameter:=True, bNewLinkedHideIfParameterMissing:=True,
+                                            bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=12)
```
Comment (lloyddewit) [inline instat/dlgOneVariableSummarise.vb:73]:
```suggestion
                                            bNewLinkedChangeToDefaultState:=True, objNewDefaultState:=iMaxSum)
```

File: instat/dlgOneVariableSummarise.vb [hunk 3]
```diff
@@ -158,7 +159,7 @@ Public Class dlgOneVariableSummarise
         clsSummariesList.AddParameter("summary_sum", Chr(34) & "summary_sum" & Chr(34), bIncludeArgumentName:=False)
 
         clsSummaryFunction.SetRCommand("summary")
-        clsSummaryFunction.AddParameter("maxsum", 7)
+        clsSummaryFunction.AddParameter("maxsum", 12)
```
Comment (lloyddewit) [inline instat/dlgOneVariableSummarise.vb:162]:
```suggestion
        clsSummaryFunction.AddParameter("maxsum", iMaxSum)
```

**Review by lloyddewit — CHANGES_REQUESTED** [review]

**Comment by Vitalis95** [comment]
@lloyddewit , I have made changes

**Review by lloyddewit — APPROVED** [review]

**Comment by lloyddewit** [comment]
@rdstern if you can test/approve, then we can merge, thanks

**Review by rdstern — APPROVED** [review]

**Pull request merged**